plasmapy/atomic/classes.py

```diff
@@ -736,26 +775,7 @@ def reduced_mass(self, other, Z=None, mass_numb=None) -> u.kg:
                 f"Unable to find the reduced mass because the mass of "
                 f"{self.particle} is not available.") from None
 
-        if isinstance(other, (str, int)):
-                other = Particle(other, Z=Z, mass_numb=mass_numb)
-
-        if isinstance(other, Particle):
-            try:
-                mass_that = other.mass.to(u.kg)
-            except MissingAtomicDataError:
-                raise MissingAtomicDataError(
-                    f"Unable to find the reduced mass because the mass of "
-                    f"{other.particle} is not available.") from None
-        else:
-            try:
-                mass_that = other.to(u.kg)
-            except Exception as exc:  # coveralls: ignore
-                raise AtomicError(
-                    f"{other} must be either a Particle or a Quantity or "
-                    f"Constant with units of mass in order to calculate "
-                    f"reduced mass.") from exc
-
-        return (mass_this * mass_that) / (mass_this + mass_that)
+        return get_reduced_mass(mass_this, other, Z=Z, mass_numb=mass_numb)
```

Comment:
First argument should probably be self.mass here

Reply:
Would it also be a good idea to pass `self.particle` since it will be used when raising `MissingAtomicDataError` when relevant code moves to module level function from [here](https://github.com/PlasmaPy/PlasmaPy/pull/265/files#diff-cc7af528697aead5e7d3524336ce958aR776)?

Reply:
Actually, just self, probably, and then make the checks in the module level functions accept particles and strings both! The particle input decorator should handle both eventually, but in the meantime, let's do it that way!

Reply:
If we directly pass `self` from `Particle.reduced_mass()` to module level `reduced_mass()`, it works when calling `Particle.reduced_mass()` method but we'll get an error when passing `ion_mass` through the same module level `reduced_mass()`:

For example:
```python
>>> from plasmapy.atomic import classes, atomic
>>> ion = atomic.ion_mass(10)
>>> classes.reduced_mass(ion, 2)
```
```
AttributeError: 'Quantity' object has no 'mass' member
```

```python
def reduced_mass(particle_type, other, Z=None, mass_numb=None) -> u.kg
    ...
     try:
         mass_this = particle_type.mass.to(u.kg)  # <-- here
     except MissingAtomicDataError:
         raise MissingAtomicDataError(
            f"Unable to find the reduced mass because the mass of "
            f"{particle_type.particle} is not available.") from None
    ...
```

How should we go about it?